gnovm/pkg/gnolang/values_string.go

```diff
@@ -94,11 +103,13 @@ func (av *ArrayValue) ProtectedString(seen *seenValues) string {
 	ss := make([]string, len(av.List))
 	if av.Data == nil {
 		for i, e := range av.List {
-			ss[i] = e.ProtectedString(seen)
+			m.GasMeter.ConsumeGas(OpCPUAssign, CPUCYCLES)
+			ss[i] = e.ProtectedString(m, seen)
```

Comment:
What is the purpose of this consumption? The elements should be consumed during their own visit.

Reply:
We have an assignement at the line below. Shouldn't we charge for it?